src/brc/sync/signals.py

```diff
@@ -0,0 +1,69 @@
+import logging
+
+from django.conf import settings
+from django.contrib.sites.models import Site
+from django.db.models.signals import post_delete, post_save
+from django.dispatch import receiver
+from django.urls import reverse
+
+from vng_api_common.models import APICredential
+from zds_client import Client, extract_params, get_operation_url
+
+from brc.datamodel.models import Besluit
+
+logger = logging.getLogger(__name__)
+
+
+class SyncError(Exception):
+    pass
+
+
+def sync(relation: Besluit, operation: str):
+    # build the URL of the besluit
+    path = reverse('besluit-detail', kwargs={
+        'version': settings.REST_FRAMEWORK['DEFAULT_VERSION'],
+        'uuid': relation.uuid,
+    })
+    domain = Site.objects.get_current().domain
+    protocol = 'https' if settings.IS_HTTPS else 'http'
+    besluit_url = f'{protocol}://{domain}{path}'
+
+    logger.info("Zaak object: %s", relation.zaak)
+    logger.info("Besluit object: %s", besluit_url)
+
+    # figure out which remote resource we need to interact with
+    client = Client.from_url(relation.zaak)
+    client.auth = APICredential.get_auth(relation.zaak)
+
+    try:
+        pattern = get_operation_url(client.schema, f'zaakbesluit_{operation}', pattern_only=True)
+    except ValueError as exc:
+        raise SyncError("Could not determine remote operation") from exc
+
+    # The real resource URL is extracted from the ``openapi.yaml`` based on
+    # the operation
+    params = extract_params(f"{relation.zaak}/irrelevant", pattern)
+
+    try:
+        operation_function = getattr(client, operation)
+        operation_function('zaakbesluit', {'besluit': besluit_url}, **params)
+    except Exception as exc:
+        logger.error(f"Could not {operation} remote relation", exc_info=1)
+        raise SyncError(f"Could not {operation} remote relation") from exc
+
+
+def sync_create(relation: Besluit):
+    sync(relation, 'create')
+
+
+def sync_delete(relation: Besluit):
```

Comment:
this suffers the same problem as DRC that the URL is different for create/delete - you need to fetch the detail URL of the relation instead of creating the object on the zaak detail URL. There was a patch in DRC by Steven which tried to shoe-horn all of this into the `sync` function, but it became so different that it's pretty much impossible to handle this cleanly.

I'd opt for: when the relation is synced with the `create`, you (should) get back the URL of the resource in the 'other API' - you could store this URL on the relation as a private/hidden field (`_zaakbesluit = models.URLField(...)` for example). The delete would then become very simple, since you can use `relation._zaakbesluit` to apply the `DELETE` to.

Reply:
`sync_create_besluit` is changed as suggested: `_zaakbesluit` field is added to `Besluit` model. 
`sync_delete_besluit` is rewritten based on _zaakbesluit value